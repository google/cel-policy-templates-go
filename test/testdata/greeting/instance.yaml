# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

apiVersion: policy.acme.co/v1
kind: greeting
metadata:
  name: seasons-greetings
selector:
  matchLabels:
    env: prod
  matchExpressions:
    - {key: "trace", operator: "DoesNotExist"}
    - {key: "debug", operator: "In", values: ["false", "justified"]}
rules:
  - greeting: "Hello"
  - farewell: "Farewell"
  - greeting: "You survived Y2K!"
    computer_greeting: WUFZIFkySyE=
    start_date: "2000-01-01T00:00:00Z"
    end_date: "2000-01-07T00:00:00Z"
    details:
      gone:
        - 1999
      next: [2038]
  - greeting: "Happy New Year's!"
    conditions:
      - description: Ring in the New Year.
        expression: >
          request.time.getMonth() == 0 &&
          request.time.getDate() == 1"
