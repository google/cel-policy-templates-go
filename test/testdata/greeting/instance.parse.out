1~# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

2~apiVersion: 3~"policy.acme.co/v1"
4~kind: 5~"greeting"
6~metadata:7~
  8~name: 9~"seasons-greetings"
10~selector:11~
  12~matchLabels:13~
    14~env: 15~"prod"
  16~matchExpressions:17~
    - 18~{19~key: 20~"trace", 21~operator: 22~"DoesNotExist"}
    - 23~{24~key: 25~"debug", 26~operator: 27~"In", 28~values: 29~[30~"false", 31~"justified"]}
32~rules:33~
  - 34~35~greeting: 36~"Hello"
  - 37~38~farewell: 39~"Farewell"
  - 40~41~greeting: 42~"You survived Y2K!"
    43~computer_greeting: 44~"WUFZIFkySyE="
    45~start_date: 46~"2000-01-01T00:00:00Z"
    47~end_date: 48~"2000-01-07T00:00:00Z"
    49~details:50~
      51~gone:52~
        - 53~1999
      54~next:55~[56~2038]
  - 57~58~greeting: 59~"Happy New Year's!"
    60~conditions:61~
      - 62~63~description: 64~"Ring in the New Year."
        65~expression: 66~>
          request.time.getMonth() == 0 &&
          request.time.getDate() == 1"
